syntax = "proto3";
package tg;

enum ELanguage {
    LN_UNDEFINED = 0;
    LN_RU = 1;
    LN_EN = 2;

    LN_OTHER = 3;
}

enum ECategory {
    NC_UNDEFINED = 0;
    NC_SOCIETY = 1;
    NC_ECONOMY = 2;
    NC_TECHNOLOGY = 3;
    NC_SPORTS = 4;
    NC_ENTERTAINMENT = 5;
    NC_SCIENCE = 6;
    NC_OTHER = 7;

    NC_ANY = 8;
    NC_NOT_NEWS = 9;
}

enum EEmbeddingKey {
    EK_UNDEFINED = 0;
    EK_CLUSTERING = 1;
}

message TEmbeddingProto {
    EEmbeddingKey Key = 1;
    repeated float Value = 2;
}

message TDocumentProto {
    string FileName = 1;
    uint32 PubTime = 2;
    uint32 Ttl = 3;

    string Title = 4;

    ELanguage Language = 5;
    ECategory Category = 6;

    repeated TEmbeddingProto Embeddings = 7;
}
